xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/Finality.java

```diff
@@ -0,0 +1,31 @@
+package org.xrpl.xrpl4j.client;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+import java.util.Optional;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableFinality.class)
+@JsonDeserialize(as = ImmutableFinality.class)
+public interface Finality {
+
+  /**
+   * Construct a builder for this class.
+   *
+   * @return An {@link ImmutableFinality.Builder}.
+   */
+  static ImmutableFinality.Builder builder() {
+    return ImmutableFinality.builder();
+  }
+
+  XrplClient.FinalityStatus finalityStatus();
```

Comment:
Small nit, but can we move `XrplClient.FinalityStatus` into it's own enum, or maybe move it into a sub-class of `Finality`?